src/any_llm/providers/anthropic/utils.py

```
@@ -157,6 +198,18 @@ def _convert_messages_for_anthropic(messages: list[dict[str, Any]]) -> tuple[str
                     "role": "user",
                     "content": [tool_result],
                 }
+            elif message["role"] == "assistant" and (thinking_block := _build_anthropic_thinking_block(message)):
+                existing_content = message.get("content", "")
+                if isinstance(existing_content, str):
+                    content_blocks = [thinking_block]
+                    if existing_content:
+                        content_blocks.append({"type": "text", "text": existing_content})
+                else:
+                    content_blocks = [thinking_block, *existing_content]
+                message = {
```

Ruff (0.15.20)

[warning] `for` loop variable `message` overwritten by assignment target

(PLW2901)